src/Hedgehog/Gen.fs

```diff
@@ -393,11 +393,16 @@ module Gen =
     let latin1 : Gen<char> =
         char '\000' '\255'
 
-    /// Generates a Unicode character, excluding invalid standalone surrogates:
-    /// '\000'..'\65535' (excluding '\55296'..'\57343')
+    /// Generates a Unicode character, excluding noncharacters('\65534', '\65535')
+    /// and invalid standalone surrogates ('\000'..'\65535' excluding '\55296'..'\57343')
     [<CompiledName("Unicode")>]
     let unicode : Gen<char> =
-        filter (not << System.Char.IsSurrogate) unicodeAll
+
+        let isNoncharacter x =
+            x = Operators.char 65534 || x = Operators.char 65535
+    
+        filter (not << isNoncharacter)
+        <| filter (not << System.Char.IsSurrogate) unicodeAll
```

Comment:
Although this would be perfectly fine in Haskell, in F# it's more common to use the forward-pipe operator, as also mentioned by Don Syme in in his 'F# Code I Love' talk(s):

```fsharp
let unicode : Gen<char> =
    let isNonCharacter x =
           x = Operators.char 65534
        || x = Operators.char 65535
    unicodeAll
    |> filter (not << isNonCharacter)
    |> filter (not << System.Char.IsSurrogate)
```

I'd also get rid of those blank lines since the part they're separating is <= 7 LOC.

Reply:
I did it. Can I have a question?  LOC means "Lines Of Code" right? Is there a special criterion in the 7 LOC limit?

Reply:
>LOC means "Lines Of Code" right

Right. https://en.wikipedia.org/wiki/Source_lines_of_code

>Is there a special criterion in the 7 LOC limit?

I don't remember where I've first read this, either in Code Complete or Clean Code (or both(?)) 

(There's also another similar 'hint', about method parameters (that they should be <= 3, otherwise passing a custom data type is warranted).)

Reply:
> I don't remember where I've first read this, either in Code Complete or Clean Code (or both(?))

It seems that I haven't met the criterion in Clean Code.

> (There's also another similar 'hint', about method parameters (that they should be <= 3, otherwise passing a custom data type is warranted).)

Yes, I have met it in  Clean Code - the function part.